src/ui/public/state_management/state_storage/_tests__/hashed_item_store.js

```diff
@@ -0,0 +1,285 @@
+              expect(touched).to.be.above(time);
+            } catch (e) {
+              return done(e);
+            }
+            done();
+          }, 1);
+        });
+
+        it('persists the item in sessionStorage', () => {
+          hashedItemStore.setItem(hash, item);
+          expect(sessionStorage.getItem(hash)).to.equal(item);
+        });
+
+        it('returns true', () => {
+          const result = hashedItemStore.setItem(hash, item);
+          expect(result).to.equal(true);
+        });
+      });
+
+      describe(`if there isn't storage space`, () => {
+        let sessionStorage;
+        let hashedItemStore;
+        let storageSizeLimit;
+        const hash = 'a';
+        const item = JSON.stringify({});
+
+        function calculateItemSize(itemCount) {
+          const itemSize = hash.length + item.length;
+          return itemSize * itemCount;
+        }
+
+        function calculateIndexSize(itemCount) {
+          const index = {};
+          for (var i = 0; i < itemCount; i++) {
+            const hash = i.toString();
+            index[hash] = {
+              hash,
+              touched: Date.now(),
+            };
+          }
+          return JSON.stringify(index).length;
+        }
+
+        beforeEach(done => {
+          sessionStorage = new StubBrowserStorage();
+
+          // Anticipate filling out storage with a persisted index (containing 2 items)
+          // and the 2 items themselves.
+          // NOTE: This is a very brittle test because it depends so heavily on the
+          // implementation of the tested code.
+          const itemCount = 2;
+          const itemsSize = calculateItemSize(itemCount);
+          const persistedIndexKeySize = HashedItemStore.PERSISTED_INDEX_KEY.length;
+          const persistedIndexValueSize = calculateIndexSize(itemCount);
+
+          storageSizeLimit = itemsSize + persistedIndexKeySize + persistedIndexValueSize;
+          sessionStorage._setSizeLimit(storageSizeLimit);
+
+          hashedItemStore = new HashedItemStore(sessionStorage);
+
+          // Add some items that will be removed.
+          hashedItemStore.setItem('1', item);
+          setTimeout(() => {
+            // Do this a little later so that this item is newer.
+            hashedItemStore.setItem('2', item);
+            done();
+          }, 1);
+        });
+
+        describe(`and the item will fit`, () => {
+          it('removes older items until the new item fits', () => {
+            hashedItemStore.setItem(hash, item);
+            expect(sessionStorage.getItem('1')).to.equal(null);
+            expect(sessionStorage.getItem('2')).to.equal(item);
+          });
+
+          it('updates the indexed item as recently touched', done => {
+            const time = Date.now();
+            setTimeout(() => {
+              try {
+                hashedItemStore.setItem(hash, item);
+                const touched = hashedItemStore._itemIndex.a.touched;
```

Comment:
Good catch! I think it's better to refer to the variable consistently instead of hard-coding it consistently, just to get the benefits of a var (var names convey meaning, you can change it in one place and update everywhere, etc). But I also think we should delete this test since it depends too much on the implementation. Instead, I think we should add a test the shows an item retrieved with `getItem` not being removed when storage space needs to be freed up.